### ISSUE TITLE
Support for board YFS-N2?

### ISSUE THREAD

**Issue by ltzen77** [RoboDurden/Hoverboard-Firmware-Hack-Gen2.x#29]
Any change any of the existing code will support this one? says YFS-N2 on the back.

![IMG_20230929_201658393](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/51126112/98545a16-1f6f-40fb-a405-46b256cd69e1)
![IMG_20230929_201643292](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/51126112/6657018c-9911-48ba-b5ac-f467bb746ad5)

**Comment by RoboDurden** [comment]
Very unlikely. 
I do not find any information about a LKS 32bit C6T8 MCU on the internet:
![grafik](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/12238841/2d01be92-d8e4-43d1-9ff8-45bfacef2614)

**Comment by ltzen77** [comment]
ah of course - with a magnifier i read:  
LKS
LKS32MC
R11C6T8
2128

**Comment by RoboDurden** [comment]
no compiler support i guess.

**Comment by AILIFE4798** [comment]
please read datasheet
[Uploading 2210201730_LINKOSEMI-LKS32MC033H6P8_C5209379.pdf…]()

**Comment by RoboDurden** [comment]
Thanks AiLife. Unfortunately i am still a beginner with this Keil IDE. So i do  not know how to add a new target.
Someone else did it with the GD32E230
And we also need a new target for the MM32SPIN05 :-/
See https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/issues/22

**Comment by AILIFE4798** [comment]
the datasheet i sent is for wrong IC
heres the correct one
[2110081130_LINKOSEMI-LKS32MC088C6T8_C2798416.pdf](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/files/13849969/2110081130_LINKOSEMI-LKS32MC088C6T8_C2798416.pdf)
i also know nothing about keil but heres the device package from manufacture website maybe itll help
[https://www.lksmcu.com/index.php/DevicePack/](url)

**Comment by AILIFE4798** [comment]
anyone have this board,do not replace the mcu with gd32,it is pin compatiable but 5v,so dont do it in any case

**Comment by RoboDurden** [comment]
You mean there is not 3.3V regulator on that layout but only the 5V regulator which directly powers the MCU ?

**Comment by AILIFE4798** [comment]
yes you can see there is no ams1117 which would be on other board

**Comment by AILIFE4798** [comment]
the mm32spin is also 5v tolerant ! please for your own good make sure the voltage before you do anything
![image](https://github.com/RoboDurden/Hoverboard-Firmware-Hack-Gen2.x/assets/142502122/3b98088c-f33a-4f3e-bca1-ed74558c5e19)

**Comment by RoboDurden** [comment]
Have you tested the mcu voltage ? If the board does not need 5V, it could go directly from the dcdc converter to 3.3V.

**Comment by AILIFE4798** [comment]
ams1117 is only rated for 12v,so it must be regulated to 5v first,and the hall sensor always need 5v,so it have to be generated

**Comment by RoboDurden** [comment]
@I-hate-2FA can give you more info on the LKS i think.
I would search the web for a LKS32 blink example.
And for sure ask https://chat.deepseek.com !

**Comment by I-hate-2FA** [comment]
Use keil or pyocd

**Comment by RoboDurden** [comment]
As I do not offer LKS32 binaries, I already assumed that he tries to compile and upload with Keil Ide. But that the st-link dongle might be the problem.

**Comment by ltzen77** [comment]
the ambition with stm 32 cube was to see so that stlink could connect at all. which failet. but ill see if any of the others will work better, thanks

**Comment by I-hate-2FA** [comment]
The dongle is not the problem, all core m devices can use it, if you used the correct device pack in keil

**Comment by I-hate-2FA** [comment]
Because mm32 and lks is not direct clone, the register location is different, you need to use different device pack
If you still can't connect, the MCU is unlocked, you need to find a way to unlock, maybe can check the mm32 as refrence

**Comment by RoboDurden** [comment]
If you want to make an autodetect binary for LKS32, you have to add another #ifdef clause in RemoteAutodetect.c that defines the available pins (digital and analog):

```
	#ifdef GD32F103	// no PF0 and PF1 on 48 pin package
		#define COUNT_PinDigital 29
		PinAD aoPin[COUNT_PinDigital] = {
				{PC13,"C13",0}	,{PC14,"C14",0}	,{PC15,"C15",0}			,
				{PA0,"A0",1}		,{PB11,"B11",0}		,{PA4,"A4",1}		,{PA5,"A5",1}	,{PA6,"A6",1} 	,
				{PA7,"A7",1}		,{PB0,"B0",1}		,{PB1,"B1",1}		,{PB2,"B2",0}	,{PB10,"B10",0}	,
				{PA1,"A1",1}	,{PB12,"B12",0}	,{PA11,"A11",0}	,{PF6,"F6",0}	,{PF7,"F7",0} 	,
				{PA12,"A12",0}	,{PA15,"A15",0}	,{PB3,"B3",0}		,{PB4,"B4",0}	,{PB5,"B5",0}	,
				{PB6,"B6",0}		,{PB7,"B7",0}		,{PB8,"B8",0}		,{PB9,"B9",0}	,{PA2,"A2",1}	,
				{PA3,"A3",1}
				};

	#else
		#define COUNT_PinDigital
		PinAD aoPin[COUNT_PinDigital] = {
				{PC13,"C13",0}	,{PC14,"C14",0}	,{PC15,"C15",0}	,{PF0,"F0",0}	,{PF1,"F1",0}		,
				{PA0,"A0",1}		,{PB11,"B11",0}		,{PA4,"A4",1}		,{PA5,"A5",1}	,{PA6,"A6",1} 	,
				{PA7,"A7",1}		,{PB0,"B0",1}		,{PB1,"B1",1}		,{PB2,"B2",0}	,{PB10,"B10",0}	,
				{PA1,"A1",1}	,{PB12,"B12",0}	,{PA11,"A11",0}	,{PF6,"F6",0}	,{PF7,"F7",0} 	,
				{PA12,"A12",0}	,{PA15,"A15",0}	,{PB3,"B3",0}		,{PB4,"B4",0}	,{PB5,"B5",0}	,
				{PB6,"B6",0}		,{PB7,"B7",0}		,{PB8,"B8",0}		,{PB9,"B9",0}	,{PA2,"A2",1}	,
				{PA3,"A3",1}
				};
	#endif
```

When adding the LKS32 target to Keil, you have to create a new define like `LKS32MCR11`

![Image](https://github.com/user-attachments/assets/e4d0d6f6-7998-4252-944e-c245e85ab0c9)

@I-hate-2FA , are the 6 bldc mosfet pins an the uart pins the same for LKS32 ?

```
	mostly used for 6 BLDC mosfet pins: B13 B14 B15 A8 A9 A10
	mostly used for USART0: B6 B7
	mostly used for USART1: A2 A3
```

@ltzen77 , for sure, do only test with a 2A constant current power supply !!!

**Comment by I-hate-2FA** [comment]
> If you want to make an autodetect binary for LKS32, you have to add another #ifdef clause in RemoteAutodetect.c that defines the available pins (digital and analog):

problem is the timer code may be different, in mm32 autodetect i do not initialize adc timer, only the necessary timer1 to drive motor



> are the 6 bldc mosfet pins an the uart pins the same for LKS32 ?

it is the same place but different wiring

![Image](https://github.com/user-attachments/assets/e5a6c324-7ffa-48c2-8213-4ae1d2af1422)

![Image](https://github.com/user-attachments/assets/ac5359a2-ff5f-4e52-a8a6-f81caa545683)

p1.4-p1.9(pb4-pb9 if we follow the naming), but it alternate between ch0hi,ch0lo,ch1hi like this, it is way better for pcb routing this way id say

it is for LKS32MC08X , i dont know which one you have
